net.go

```diff
@@ -0,0 +1,89 @@
+package gossipcache
+
+import (
+	"net"
+	"net/netip"
+)
+
+// GetInterfacesNames returns the list of interfaces,
+// considering an optional exclusion list
+func GetInterfacesNames(except ...string) ([]string, error) {
+	var out []string
+
+	s, err := net.Interfaces()
+	if err != nil {
+		return out, err
+	}
+
+	for _, ifi := range s {
+		name := ifi.Name
+
+		for _, nope := range except {
+			if name == nope {
+				name = ""
+				break
+			}
+		}
+
+		if name != "" {
+			out = append(out, name)
+		}
+	}
+
+	return out, nil
+}
+
+// GetIPAddress tries to return the address we are most likely to use
+// to communicate with the network
+func GetIPAddress(ifaces ...string) (netip.Addr, error) {
+	// TODO: consider networks and return "best"
+	addrs, err := GetIPAddresses(ifaces...)
+	if len(addrs) > 0 {
+		return addrs[0], nil
+	}
+	return netip.Addr{}, err
+}
+
+// GetIPAddresses returns the list of IP Addresses,
+// optionally considering only the given interfaces
+func GetIPAddresses(ifaces ...string) ([]netip.Addr, error) {
+	var out []netip.Addr
+
+	if len(ifaces) == 0 {
+		var err error
+
+		ifaces, err = GetInterfacesNames()
+		if err != nil {
+			return out, err
+		}
+	}
+
+	for _, name := range ifaces {
+		ifi, err := net.InterfaceByName(name)
+		if err != nil {
+			return out, err
+		}
+
+		addrs, err := ifi.Addrs()
+		if err != nil {
+			return out, err
+		}
+
+		for _, addr := range addrs {
+			var s []byte
+
+			switch v := addr.(type) {
+			case *net.IPAddr:
+				s = v.IP
+			case *net.IPNet:
+				s = v.IP
+			}
+
+			if ip, ok := netip.AddrFromSlice(s); ok {
```

Comment:
Well, if I want to be nitpicking (which I don't) I would say that there are 3 possible results that come from AddrFromSlice (IPv4 if the slice length is 4) and either a IPv6 mapped IPv4 or a standard IPv6 if the slice length is 16. The thing is that you cannot actually distinguish between the later two options easily, the question is do we care? at this point I think we do not care so I will approve.

Reply:
I think those are supposed to be treated as IPv6